Yona/Yona/Others/SMSValidPasscodeLogin/Login/LoginViewController.swift

```diff
@@ -206,15 +204,17 @@ extension LoginViewController: CodeInputViewDelegate {
         let passcode = KeychainManager.sharedInstance.getPINCode()
         if code ==  passcode {
             authenticatedSuccessFully()
+            UserDefaults.standard.set(0, forKey: YonaConstants.nsUserDefaultsKeys.numberOfLoginAttempts) // resetting number of login attempts to 0
         } else {
             errorLabel.isHidden = false
             self.codeInputView.clear()
-            if loginAttempts == maxAttempts {
+            var loginAttempts:Int = UserDefaults.standard.integer(forKey: YonaConstants.nsUserDefaultsKeys.numberOfLoginAttempts)
+            if  loginAttempts == maxAttempts {
```

Comment:
If you put the retrieval of the user default in a function, you don't need the local variable ``loginAttempts`` but you can just do this:
```suggestion
            if  getFailedLoginAttempts() + 1 == maxAttempts {
```

Reply:
Updated code accordingly.